### DISCUSSION TITLE
VS code question

### DISCUSSION THREAD

**Discussion by arvobrendon** [coder/code-server#3531] [Q&A] [answered]
Hello all users ,
I’m new here quick question 

- Am I allowed to start a new project python on vs code browsers ( if yes where does the file goes )
- can I host this in shared host like c panel

**Comment by jsjoeio** [comment, marked as answer]
Yes! You can put the file anywhere. Here's a quick demo I made for you:

## Demo 

https://user-images.githubusercontent.com/3806031/120683399-6c673e80-c452-11eb-9a10-d0191439e54c.mov

If you need a more in-depth tutorial, check out these resources:
- [Getting Started with Python in VS Code](https://code.visualstudio.com/docs/python/python-tutorial)
- [Scientific Computing with Python (freeCodeCamp)](https://www.freecodecamp.org/learn/scientific-computing-with-python/)

## Hosting

Checkout our `deploy-code-server` guides [here](https://github.com/cdr/deploy-code-server). They will walk you through hosting on various providers.

**Comment by arvobrendon** [comment]
Sorry what I mean was 
Create virtual environment

and last question 
Can I locate the file on my iCloud document ?

**Reply by jsjoeio** [reply]
Ah...Hmm... it's been a while since I've written any Python or used a virtual environment. As long as you have the  dependencies installed on your remote machine running code-server, there's no reason it shouldn't work. 

In case you need it, this [seems helpful](https://docs.python.org/3/library/venv.html).

> Can I locate the file on my iCloud document ?

Assuming you're running code-server on an Apple device, you should be able to open the `~/Library/Mobile Documents` or wherever your iCloud files live in code-server. [Link](https://www.macworld.com/article/223499/how-to-access-icloud-files-from-your-mac.html)

**Reply by arvobrendon** [reply]
Hello there 
When I try to open my folder on documents or etc it says
(Bash: cd: /home/coder/Documents: no such file or directory

**Reply by jsjoeio** [reply]
It sounds like you looked in the `/home/coder` for your `Documents` folder but it's not there. If you're on a mac, you might be looking for something like `/Users/<username>/Documents`. That's how it is on mine at least. Hope that helps

**Comment by arvobrendon** [comment]
Thank you

**Comment by arvobrendon** [comment]
I hope code-server can locate file locally on ipad 
For now I will use koder 

Because now I can only access file through github not locally

anyway please add pip into it

**Reply by jsjoeio** [reply]
You can install pip yourself! We don't limit the dependencies you install locally in your environment

**Comment by arvobrendon** [comment]
Is there a way to access /home/coder on iPad 
I tried install pip doesn’t work in it

**Reply by jsjoeio** [reply]
if you run code-server on your iPad, you should be able to.

Run `code-server --link` and then open that link on your iPad.


> I tried install pip doesn’t work in it

Sorry, I wish I could help but I think you'll just have to read [the pip documentation](https://pip.pypa.io/en/stable/installing/) and see if you can figure it out.
